test/e2e/workload_api_test.go

```diff
@@ -204,8 +204,7 @@ func TestDockerUpgradeKubernetes123to124WorkloadClusterScaleWorkerNodesAPI(t *te
 				api.WithManagementCluster(managementCluster.ClusterName),
 				api.WithControlPlaneCount(1),
 				api.RemoveAllWorkerNodeGroups(), // This gives us a blank slate
-				api.WithWorkerNodeGroup("worker-0", api.WithCount(2)),
-				api.WithWorkerNodeGroup("worker-1", api.WithCount(1)),
+				api.WithWorkerNodeGroup("worker-0", api.WithCount(1)),
```

Comment:
Why are we changing this to remove the scale down case?

Reply:
This one was changed to primarily test the k8s upgrade, so I figured scaling up is all that's needed. The new test added tests scaling up and down, as well as adding / remove worker node groups,